lucid/misc/redirected_relu_grad.py

```diff
@@ -0,0 +1,120 @@
+# Copyright 2018 The Lucid Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+"""Redirected ReLu Gradient Overrides
+
+When visualizing models we often[0] have to optimize through ReLu activation
+functions. Where accessing pre-relu tensors is too hard, we use these
+overrides to allow gradient to flow back through the ReLu—even if it didn't
+activate ("dead neuron") and thus its derivative is 0.
+
+Usage:
+```python
+from lucid.misc.gradient_override import gradient_override_map
+from lucid.misc.redirected_relu_grad import redirected_relu_grad
+
+with gradient_override_map({'Relu': redirected_relu_grad}):
+  model.import_graph(…)
+```
+
+Discussion:
+ReLus block the flow of the gradient during backpropagation when their input is
+negative. ReLu6s also do so when the input is larger than 6. These overrides
+change this behavior to allow gradient pushing the input into a desired regime
+between these points.
```

Comment:
Clarify that this only happens when the gradient would be zero if we didn't intervene?

Reply:
ACK